plugins/basic/nexus-repository-nuget/src/main/java/com/sonatype/nexus/repository/nuget/internal/security/NugetApiKeyEventInspector.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package com.sonatype.nexus.repository.nuget.internal.security;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+//import org.sonatype.nexus.events.EventSubscriber;
+import org.sonatype.nexus.security.UserPrincipalsExpired;
+
+import com.google.common.eventbus.AllowConcurrentEvents;
+import com.google.common.eventbus.Subscribe;
+import org.apache.shiro.subject.SimplePrincipalCollection;
+
+/**
+ * Ensures that API keys for deleted users are removed.
+ *
+ * TODO: Ensure this receives events.
+ */
+@Named
+@Singleton
+final class NugetApiKeyEventInspector
+    //implements EventSubscriber
```

Comment:
why not impls here?  unless this is otherwise registered this event handler is a nope.

Could also consider adding an event handler to the store itself instead of having another adapter here?